onto-viewer-core/src/main/java/org/edmcouncil/spec/ontoviewer/core/utils/StringUtils.java

```diff
@@ -17,11 +28,21 @@ public static String getFragment(IRI iri) {
     if (iriString.contains(AXIOM_PATTERN)) {
       return iriString.substring(iriString.lastIndexOf(".") + 1);
     }
-    String iriFragment = iri.getFragment();
-    if (iriFragment.isEmpty() || iriFragment.isBlank()) {
-      return iriString.substring(iriString.lastIndexOf("/") + 1).replace("#", "");
+    String result = null;
+    if (iriString.endsWith("#")) {
+      Matcher matcher = IRI_FRAGMENT_BEFORE_HASH_PATTERN.matcher(iriString);
+      while (matcher.find()) {
+        String match = matcher.group();
+        result = IRI_FRAGMENT_PATTERN.matcher(match).group();
+      }
+    } else {
+      Matcher matcher = IRI_FRAGMENT_AFTER_HASH_PATTERN.matcher(iriString);
+      while (matcher.find()) {
+        String match = matcher.group();
+        result = IRI_FRAGMENT_PATTERN.matcher(match).group();
+      }
```

Comment:
I do not like this solution with patterns, it still uses the methods of splitting the text not explicitly in its methods,
Patterns written by us may be inaccurate and partially wrong, cutting the text earlier was safer in my opinion than this solution. 
There we defined the end and beginning of the cut and it did not matter what was in between, here we define that they should be alphanumeric characters.
I would prefer to go back to the 'split' version here. 
@trojczak do you see any other solution?

Reply:
There was a loading bug. I would have to make another pattern. Pattern that would handle:  #, /, -, etc. 
This is pointless because it is an endless work. Using split I could do it with 6 lines. Just like in the beginning.
Continue the work, corrections?
@trojczak